src/rez/solver.py

```diff
@@ -821,6 +844,7 @@ def _split(i_entry, n_variants, common_fams=None):
             for j, variant in enumerate(entry.variants):
                 fams = fams & variant.request_fams
                 if not fams:
+                    assert prev is not None
```

Comment:
Will there be a cost to this assert? Same with the other asserts in this module.